lib/octicons_react/script/build.js

```diff
@@ -52,6 +44,55 @@ import React from 'react'
 
 ${icons.map(({code}) => code).join('\n')}
 
+function closestNaturalHeight(naturalHeights, height) {
+  return naturalHeights
+    .map(naturalHeight => parseInt(naturalHeight, 10))
+    .reduce(
+      (acc, naturalHeight) => (naturalHeight <= height ? naturalHeight : acc),
+      naturalHeights[0]
+    )
+}
+
+const sizeMap = {
+  small: 16,
+  medium: 32,
+  large: 64
+}
+
+function getSvgProps({
+  ariaLabel,
+  className,
+  size,
+  verticalAlign,
+  svgDataByHeight
+}) {
+  const height = sizeMap[size] || size
+  const naturalHeight = closestNaturalHeight(
+    Object.keys(svgDataByHeight),
+    height
+  )
+  const naturalWidth = svgDataByHeight[naturalHeight].width
+  const width = height * (naturalWidth / naturalHeight)
+  const path = svgDataByHeight[naturalHeight].path
+
+  return {
+    'aria-hidden': ariaLabel ? 'false' : 'true',
+    'aria-label': ariaLabel,
+    role: 'img',
+    className,
+    viewBox: \`0 0 $\{naturalWidth} $\{naturalHeight}\`,
+    width,
+    height,
+    fill: 'currentColor',
+    style: {
+      display: 'inline-block',
+      userSelect: 'none',
+      verticalAlign
+    },
+    dangerouslySetInnerHTML: { __html: path }
+  }
+}
```

Comment:
This is a lot of static code in a template string. I'll pull it out into a file.